code/modules/reagents/reagent_containers/syringes.dm

```diff
@@ -157,16 +157,17 @@
 
 				reagents.reaction(target, INGEST)
 			spawn(5)
-				target.add_fingerprint(user)
-				var/datum/reagent/blood/B = locate() in src.reagents.reagent_list
-				var/trans
+				var/datum/reagent/blood/B
+				for(var/datum/reagent/blood/d in src.reagents.reagent_list)
+					B = d
+					break
 				if(B && istype(target,/mob/living/carbon))
 					var/mob/living/carbon/C = target
 					C.inject_blood(src,5)
 				else
-					trans = src.reagents.trans_to(target, amount_per_transfer_from_this)
-				user << "<span class='notice'>You inject [trans] unit\s of the solution. [src] now contains [reagents.total_volume] unit\s.</span>"
-				if(reagents.total_volume <= 0 && mode == SYRINGE_INJECT)
+					src.reagents.trans_to(target, amount_per_transfer_from_this)
+				user << "<span class='notice'>You inject 5 units of the solution. The syringe now contains [src.reagents.total_volume] units.</span>"
+				if (reagents.total_volume <= 0 && mode==SYRINGE_INJECT)

```

Comment:
You doing ´´´if (´´´ after having made https://github.com/tgstation/-tg-station/pull/7404 is kinda fully.